client/containers/workflow-pending/constants.js

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2021 Uber Technologies Inc.
+//
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {
+  PENDING_TASK_TYPE_ACTIVITY,
+  PENDING_TASK_TYPE_CHILD_WORKFLOW,
+} from '../workflow/constants';
+
+export const PENDING_TASK_FILTER_TO_EMPTY_MESSAGE_MAP = {
+  all: 'No pending tasks',
+  activities: 'No pending activities',
+  children: 'No pending child workflows',
+};
+
+export const PENDING_TASK_KVPS_EXCLUDE_KEYS = [
+  'pendingTaskId',
+  'pendingTaskType',
+  'pendingTaskTypeDisplay',
+];
+
+export const PENDING_TASK_TYPE_TO_ID_MAP = {
+  [PENDING_TASK_TYPE_ACTIVITY]: 'activityID',
+  [PENDING_TASK_TYPE_CHILD_WORKFLOW]: 'initiatedID',
```

Comment:
maybe just because i'm not familiar with the code patterns here:  why an array as a key?

Reply:
it's using a constant value as the key for the parameter.

so for this particular key:
`PENDING_TASK_TYPE_CHILD_WORKFLOW = 'PENDING_TASK_TYPE_CHILD_WORKFLOW'` (see `client/containers/workflow/constants.js`)

this will then evaluate to:
```
PENDING_TASK_TYPE_TO_ID_MAP = {
  PENDING_TASK_TYPE_CHILD_WORKFLOW: 'initiatedID'
};
```

so when using this map and the task type = PENDING_TASK_TYPE_CHILD_WORKFLOW it will resolve to "initiatedID".

i.e.
```
const pendingTask = {
  pendingTaskType: 'PENDING_TASK_TYPE_CHILD_WORKFLOW',
  initiatedID: '1234',
  ...params
};

const idField = PENDING_TASK_TYPE_TO_ID_MAP[pendingTask.pendingTaskType];
console.log(idField); // logs "initiatedID"
console.log(pendingTask[idField]) // logs "1234"
```

Reply:
unless `[const]` has some special behavior (does it?  I'm not up-to-date on modern js), this is still `{[const]:'initiatedID'}` which means `pendingTaskType` needs to be `[const]`, not just `const`.

I assume it is, but that's still kinda weird.  does it simplify code somewhere?  I see these used with "filter" in a number of places, which seems to imply it's reducing a larger list, but it doesn't seem to actually *process as a list*, so e.g. `[PENDING_TASK_TYPE_CHILD_WORKFLOW, SOME_OTHER_CONST]` would fail to match anything and no-op[

Reply:
no typing in javascript, only in typescript. These will resolve to strings. Within the context of an object, using square brackets will use a variable in place.

i.e.
```
const myVariableFieldName = 'hello';
const myObject = {
  hello: 'world',
};
console.log(myObject[hello]); // logs "world"
```

The idea behind getter-types, action-types, mutation-types etc. is to allow autocompletion in IDEs when searching for getters / actions / mutations available in a container or in a different container.

`const` is equivalent to `var` with the exception that it cannot be mutated after its defined.
i.e.
```
const myConstant = '1234';
myConstant = '5678' // will result in javascript throwing an exception of some kind
```

Reply:
(chatted IRL) 
Ah.  It's an ES6 syntax for using variables as keys.  Makes sense then 👍